### PR TITLE
Adds new Golden Deep Ornate Energy Pistol and gives it to Consular

### PR DESCRIPTION
**Pull request by Ben10083**
New Golden Deep Energy Pistol with one unique mode: Disorient
Disorient: weaker stun but applies effects such as slurring and dizzy

TODO:
- [x] Name/desc/lore misc. For item
- [x] Weapon Sprite (credit to @NobleRow 
- [x] Add confusion?

image

### CODE REVIEW

**Comment by Ben10083** [comment]
!wip

**Comment by Ben10083** [comment]
Going to have the SCC beam used for disorient. A dedicated projectile sprite may be made in the future if my comrade is able to.

**Comment by Ben10083** [comment]
!review

**Comment by Ben10083** [comment]
image

**Comment by DreamySkrell** [comment]
Should not be using the SCC projectile sprite

**Comment by DreamySkrell** [comment]
Does the disorienting work on both humans and IPCs? Reading the code it applies the effects to `/mob/living/carbon/human` which includes IPCs. I don't think the effects would make sense to apply to IPCs (if they do).

File: code/modules/projectiles/projectile/beams.dm [hunk 1]
```diff
@@ -222,6 +222,31 @@
...
+
+	muzzle_type = /obj/effect/projectile/muzzle/disabler
+	tracer_type = /obj/effect/projectile/tracer/disabler
+	impact_type = /obj/effect/projectile/impact/disabler
+
+/obj/item/projectile/beam/disorient/on_hit(var/atom/target, var/blocked = 0)
+	if(ishuman(target) && blocked < 100 && !issilicon(target) && !isipc(target)) //Make them trip
+		var/mob/living/carbon/human/H = target
+		if(H.druggy < 75)
+			H.druggy += 15
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/projectile/beams.dm:242]: this should be capped at 75
`H.druggy = min(H.druggy + 15, 75)`

File: code/modules/projectiles/projectile/beams.dm [hunk 2]
```diff
@@ -222,6 +222,31 @@
...
+		if(H.dizziness < 250) //too much dizzy is BAD. we add a cap here
+			H.make_dizzy(50)
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/projectile/beams.dm:244]: as above, check that it doesn't go over 250

File: code/modules/projectiles/projectile/beams.dm [hunk 3]
```diff
@@ -222,6 +222,31 @@
...
+		H.confused = 5
+		if(H.slurring < 75)
+			H.slurring += 15
```
Comment (NonQueueingMatt) [inline code/modules/projectiles/projectile/beams.dm:247]: as above

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 4]
```diff
@@ -177,6 +177,32 @@
 		list(mode_name="smite", projectile_type=/obj/item/projectile/beam/pistol/hegemony, modifystate="hegemony_pistol", fire_sound='sound/weapons/laser1.ogg')
 		)
 
+/obj/item/gun/energy/pistol/goldendeep
+	name = "golden deep ornate energy pistol"
+	desc = "An intricate golden energy pistol, engraved with a quality unlike no other. If you had to describe this pistol in one word, it would be <b>expensive.</b>"
+	desc_extended = "Possessed by only the most affluent affiliates of the Golden Deep, this ornate energy pistol covered in gold is the result of a commission, which had the artistic quality of Ultra Maz combined with the scientific technology acquired by the Estriconian to produce a weapon that many outsiders see as a firearm more valuable than the assets it was made to protect."
+	icon = 'icons/obj/guns/goldendeep_ornatepistol.dmi'
+	icon_state = "golden_deep_ornatepistolstun100"
+	item_state = "golden_deep_ornatepistolstun100"
```
Comment (FluffyGhoster) [inline code/modules/projectiles/guns/energy/nuclear.dm:186]: The file already says it's golden deep, there's no point in the icon state repeating this info

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 5]
```diff
@@ -177,6 +177,32 @@
...
+	max_shots = 10
+	fire_delay = 3
+	can_turret = FALSE
+	secondary_projectile_type = /obj/item/projectile/beam/pistol
+	secondary_fire_sound = 'sound/weapons/laser3.ogg'
+	can_switch_modes = TRUE
+
+	projectile_type = /obj/item/projectile/beam/disorient
+	origin_tech = list(TECH_COMBAT = 4, TECH_MAGNET = 3)
+	modifystate = "golden_deep_ornatepistolstun"
```
Comment (FluffyGhoster) [inline code/modules/projectiles/guns/energy/nuclear.dm:199]: As above

**Review by FluffyGhoster — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**